### PR TITLE
Feature/notes duration recognition two methods

### PR DESCRIPTION
**Pull request by JakubBilski**

### CODE REVIEW

File: src/meter_recognition.py [hunk 1]
```diff
@@ -1,14 +1,146 @@
...
+    relation_log = log2(relation)
+    minimized_values = [abs(arl-relation_log) for arl in allowed_fractions_log]
+    # print(minimized_values)
+    for dotted_note_index in range(1, len(minimized_values), 2):
+        minimized_values[dotted_note_index] += DOTTED_NOTES_DISCRIMINATOR
+    return min(range(len(allowed_fractions_log)),
+                            key=minimized_values.__getitem__)
+
+
+def transform_beat_fractions_into_duration_signatures(sounds, meter):
```
Comment (JakubBilski) [inline src/meter_recognition.py:25]: _into_rhytmic_values

File: src/music/sound.py [hunk 2]
```diff
@@ -1,21 +1,46 @@
...
-    def __init__(self, note, timestamp=None, duration=None):
+    def __init__(self, note=None, timestamp=None, duration=None,
+                 duration_signature=None, symbol=None):
         self.note = note
+        if symbol is not None:
+            self.note = REVERSE_SYMBOLS[symbol]
         self.timestamp = timestamp
         self.duration = duration
+        self.duration_signature = duration_signature
+        self.beat_fraction = None
```
Comment (JakubBrojacz) [inline src/music/sound.py:31]: Remove `duration_signature` and `beat_fraction`. 
Use only rhythmic_value (type string, previous duration_signature)

File: src/music/sound.py [hunk 3]
```diff
@@ -1,21 +1,46 @@
...
+
 class Sound:
-    def __init__(self, note, timestamp=None, duration=None):
+    def __init__(self, note=None, timestamp=None, duration=None,
+                 duration_signature=None, symbol=None):
         self.note = note
+        if symbol is not None:
+            self.note = REVERSE_SYMBOLS[symbol]
         self.timestamp = timestamp
         self.duration = duration
```
Comment (JakubBrojacz) [inline src/music/sound.py:29]: Change to `duration_ms` to match documentation

File: src/meter_recognition.py [hunk 4]
```diff
@@ -1,14 +1,146 @@
...
+def update_sounds_with_beat_fractions_compare_adjacent(sounds, meter):
+    start_index = min(range(len(sounds)), key=[abs(s.duration-meter) for s in sounds].__getitem__)
+    if sounds[start_index].duration/meter > 1.5 or sounds[start_index].duration/meter < 0.75:
+        raise("Wow, there was no note with duration similar to the beat")
+    sounds[start_index].beat_fraction = 1.0
+    allowed_fractions = [16.0, 12.0, 8.0, 6.0, 4.0, 3.0, 2.0, 1.5, 1.0, 0.75, 0.5, 0.375, 0.25, 0.1875, 0.125, 0.09375, 0.0625]
+    allowed_fractions_log = [log2(ar) for ar in allowed_fractions]
+    for i in range(start_index+1, len(sounds)):
+        best_match_index = find_best_fraction_index(sounds[i], sounds[i-1], allowed_fractions_log)
+        sounds[i].beat_fraction = allowed_fractions[best_match_index]
```
Comment (JakubBilski) [inline src/meter_recognition.py:128]: This is some relict of an old approach, beat_fraction should not be a Sound property - it is used only locally and introduce terrible ambiguity later on

File: src/music/sound.py (hunk 2 above)
Comment (JakubBilski) [inline src/music/sound.py:31]: Should not be a property - it is used only locally in places there it's used, should be deduced from metrum

Comment (JakubBilski) [reply src/music/sound.py:31]: as Kuba mentioned above, should use only rhytmic_value

File: src/main_full.py [hunk 5]
```diff
@@ -19,36 +25,52 @@ def parse_args():
...
 
     print("Chords:")
     print("\n".join([f"{chord.timestamp:.3f}: {chord}\t{chord.duration:.3f}" for chord in chords]))  # noqa
 
     music_synthesis.midi_from_sounds(sounds, 'sounds')
     music_synthesis.midi_from_chords(chords, 'chords')
 
-    sound1 = AudioSegment.from_file("sounds.wav")
-    sound2 = AudioSegment.from_file("chords.wav")
+    sound1 = AudioSegment.from_file("sounds.mid")
```
Comment (JakubBilski) [inline src/main_full.py:60]: Throw out exporting until we find a working way to do it

File: src/main_full.py [hunk 6]
```diff
@@ -19,36 +25,52 @@ def parse_args():
     args = parser.parse_args()
     return args
 
-
 if __name__ == "__main__":
     args = parse_args()
-    sounds = sounds_generation.get_sounds_from_file(args.input)
+    file = args.input
+    sounds = sounds_generation.get_sounds_from_file(file)
+    sounds = sounds_manipulation.change_tonation(sounds, 2)
```
[the remark this answers is not in the document]
Comment (JakubBilski) [reply src/main_full.py]: Commented

File: src/main_full.py [hunk 7]
```diff
@@ -19,36 +25,52 @@ def parse_args():
...
-    meter = meter_recognition.get_meter(args.input, sounds)
-    sounds = meter_recognition.update_sounds(meter, sounds)
-    print(f"Metrum: {meter}")
+    print(f"Meter: {meter}")
 
     print("Sounds:")
-    print("\n".join([f"{sound.timestamp:.3f}: {sound.symbol}\t{sound.duration:.3f}" for sound in sounds]))  # noqa
+    print("\n".join([f"{sound.timestamp:.3f}: {sound.symbol}\t{sound.duration:.3f} ({sound.duration_signature})" for sound in sounds]))  # noqa
 
     tonation = tonations_generation.get_tonations_from_sounds(sounds)
```
[the remark this answers is not in the document]
Comment (JakubBilski) [reply src/main_full.py:49]: Done

File: src/meter_recognition.py [hunk 8]
```diff
@@ -1,14 +1,146 @@
...
+from math import log2
 
+from music import Sound
 
-def get_meter(filename, sounds):
-    # TODO use sounds?
-    y, sr = librosa.load(filename)
-    tempo, _ = librosa.beat.beat_track(y=y, sr=sr)
-    return 60.0/tempo
+DOTTED_NOTES_DISCRIMINATOR = 0.0 #0.0 - no discrimination, 1.0 - no dotted notes (not linear!)
```
[the remark this answers is not in the document]
Comment (JakubBilski) [reply src/meter_recognition.py]: Comment added

File: src/tests/test_data_clear.txt [hunk 9]
```diff
@@ -0,0 +1,309 @@
+data/other_rec/gdy_sliczna_panna_F.wav
```
[the remark this answers is not in the document]
Comment (JakubBilski) [reply src/tests/test_data_clear.txt]: test_data_clear was a temporary file for Ania, I deleted it and changed to test_data.txt in test cases generation

File: src/tests/tonation_from_sounds_test.py [hunk 10]
```diff
@@ -10,16 +10,17 @@ def parse_args():
     parser = argparse.ArgumentParser(
```
[the remark this answers is not in the document]
Comment (JakubBilski) [reply src/tests/tonation_from_sounds_test.py:12]: Changed. I also change tonations_generation.py to tonation_recognition.py

File: src/music/sound.py (hunk 3 above)
Comment (JakubBilski) [reply src/music/sound.py]: Done

File: src/meter_recognition.py (hunk 4 above)
Comment (JakubBilski) [reply src/meter_recognition.py]: Fixed

File: src/music/sound.py (hunk 2 above)
Comment (JakubBilski) [reply src/music/sound.py:31]: Fixed

File: src/main_full.py (hunk 5 above)
Comment (JakubBilski) [reply src/main_full.py]: Commented out

File: src/music/sound.py (hunk 2 above)
Comment (JakubBilski) [reply src/music/sound.py]: Done

File: src/meter_recognition.py (hunk 1 above)
Comment (JakubBilski) [reply src/meter_recognition.py]: Done

**Review by JakubBrojacz — APPROVED** [review]

**Pull request merged**